### PR TITLE
[intel-mkl] Update description field

### LINKED ISSUE #18896: [intel-mkl] build failure using GitHub Actions (MSBuild based project)
**Host Environment**
- OS: Windows via GitHub Action ([msvc_build.yml](https://github.com/rogerfraser/DynAdjust/blob/master/.github/workflows/msvc_build.yml))
  ```yaml
  env:
    BUILD_CONFIGURATION: Release
    ARCH: x64

  jobs:
    build:
      runs-on: windows-latest
  ...
  
  ```
  
- Compiler: MSBuild (supplied by GitHub Actions virtual environments)
  ```yaml
  steps:
  ...

  - name: Add MSBuild to PATH
    uses: microsoft/setup-msbuild@v1.0.2  
    
  ...
  
  ```

**To Reproduce**
```yaml
- name: Install vcpkg prerequisites
  shell: cmd
  if: steps.cache.outputs.cache-hit != 'true'
  run: |
    vcpkg update
    vcpkg install intel-mkl --triplet=${{env.ARCH}}-windows
    vcpkg ...
    vcpkg integrate install
    vcpkg list
    
  ...
  
  ```

**Failure logs** 
- See full log [here](https://github.com/rogerfraser/DynAdjust/runs/3034571347?check_suite_focus=true).

```txt
Run vcpkg update
  vcpkg update
  vcpkg install intel-mkl --triplet=x64-windows
  shell: C:\Windows\system32\cmd.EXE /D /E:ON /V:OFF /S /C "CALL "{0}""
  env:
    SOLUTION_FILE_PATH: ./dynadjust/dynadjust_1_01_00.sln
    BUILD_CONFIGURATION: Release
    ARCH: x64
    VCPKG_INSTALL_FOLDER: c:/vcpkg/installed
    CS_XSD_INSTALL_FOLDER: xsd/xsd-4.0.0-i686-windows/libxsd
Using local portfile versions. To update the local portfiles, use `git pull`.
No packages need updating.
Computing installation plan...
The following packages will be built and installed:
    intel-mkl[core]:x64-windows -> 2020.0.0
Detecting compiler hash for triplet x64-windows...
Could not locate cached archive: C:\Users\runneradmin\AppData\Local\vcpkg\archives\f8\f8e99c4db6108860ec37bb79c9bbdbbc05a34887cc655ec8fcd3a7400f38d72b.zip
Starting package 1/1: intel-mkl:x64-windows
Building package intel-mkl[core]:x64-windows...
CMake Error at ports/intel-mkl/portfile.cmake:15 (message):
  Could not find MKL.  Before continuing, please download and install MKL
  (20200000 or higher) from:

      https://registrationcenter.intel.com/en/products/download/3178/

  

  Also ensure vcpkg has been rebuilt with the latest version (v0.0.104 or
  later)
Call Stack (most recent call first):
  scripts/ports.cmake:139 (include)


Error: Building package intel-mkl:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: intel-mkl:x64-windows
  Vcpkg version: 2021-05-05-b110959e13f562c2a17d7bd9e352e2dd8d6dc603

Additionally, attach any relevant sections from the log files above.
Error: Process completed with exit code 1.
```

**Additional context**
- Successful build of the repo requires other packages (e.g. boost-*, 7zip, xerces-c, etc.).   Installation of these packages (e.g. `vcpkg install boost-process ... --triplet=${{env.ARCH}}-windows`) is successful. 
- Caching was originally applied, then removed, but neither cases were successful.
- Full yaml [here](https://github.com/rogerfraser/DynAdjust/blob/master/.github/workflows/msvc_build.yml).

### PR DESCRIPTION
**Pull request by NancyLi1013**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #18896

Add the information that users need to install MKL manually to `description `field.

`Note that please download and install MKL from https://registrationcenter.intel.com/en/products/download/3178/ manually before using it in vcpkg.`

Note: No feature needs to test.

### CODE REVIEW

**Comment by Neumann-A** [comment]
Probably:
https://software.intel.com/content/www/us/en/develop/tools/oneapi/base-toolkit/download.html?operatingsystem=window&distributions=webdownload&options=offline

File: ports/intel-mkl/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "intel-mkl",
+  "version": "2020.0.0",
+  "port-version": 2,
+  "description": "Intel® Math Kernel Library (Intel® MKL) accelerates math processing routines, increases application performance, and reduces development time on Intel® processors. Note that please download and install MKL from https://registrationcenter.intel.com/en/products/download/3178/ manually before using it in vcpkg."
```
Comment (PhoebeHui) [inline ports/intel-mkl/vcpkg.json:5]:
```suggestion
  "description": "Intel® Math Kernel Library (Intel® MKL) accelerates math processing routines, increases application performance, and reduces development time on Intel® processors.  Due to the complexity involved, this package doesn't install MKL. It instead verifies that MKL is installed. Please download and install MKL from https://registrationcenter.intel.com/en/products/download/3178/ manually before using it in vcpkg."
```

**Comment by NancyLi1013** [comment]
> Probably:
> https://software.intel.com/content/www/us/en/develop/tools/oneapi/base-toolkit/download.html?operatingsystem=window&distributions=webdownload&options=offline

Thanks for your suggestion. @Neumann-A 

Since https://registrationcenter.intel.com/en/products/download/3178/ is also used in portfile.cmake, it would be better to keep the consistent with this in vcpkg.json.

**Review by PhoebeHui — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**